crates/bevy_ecs/macros/src/fetch.rs

```diff
@@ -0,0 +1,577 @@
+
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+
+    let struct_name = ast.ident.clone();
+    let item_struct_name = Ident::new(&format!("{}Item", struct_name), Span::call_site());
+    let read_only_item_struct_name = if fetch_struct_attributes.is_mutable {
+        Ident::new(&format!("{}ReadOnlyItem", struct_name), Span::call_site())
+    } else {
+        item_struct_name.clone()
+    };
+    let fetch_struct_name = Ident::new(&format!("{}Fetch", struct_name), Span::call_site());
+    let state_struct_name = Ident::new(&format!("{}State", struct_name), Span::call_site());
+    let read_only_fetch_struct_name = if fetch_struct_attributes.is_mutable {
+        Ident::new(&format!("{}ReadOnlyFetch", struct_name), Span::call_site())
+    } else {
+        fetch_struct_name.clone()
+    };
+    let fetch_associated_type = Ident::new("Fetch", Span::call_site());
+    let read_only_fetch_associated_type = Ident::new("ReadOnlyFetch", Span::call_site());
+
+    let fields = match &ast.data {
+        Data::Struct(DataStruct {
+            fields: Fields::Named(fields),
+            ..
+        }) => &fields.named,
+        _ => panic!("Expected a struct with named fields"),
+    };
+
+    let mut ignored_field_attrs = Vec::new();
+    let mut ignored_field_visibilities = Vec::new();
+    let mut ignored_field_idents = Vec::new();
+    let mut ignored_field_types = Vec::new();
+    let mut field_attrs = Vec::new();
+    let mut field_visibilities = Vec::new();
+    let mut field_idents = Vec::new();
+    let mut field_types = Vec::new();
+    let mut fetch_init_types = Vec::new();
+
+    let (world_lifetime, fetch_lifetime) = match (&world_lifetime_param, &fetch_lifetime_param) {
+        (GenericParam::Lifetime(world), GenericParam::Lifetime(fetch)) => {
+            (&world.lifetime, &fetch.lifetime)
+        }
+        _ => unreachable!(),
+    };
+
+    for field in fields.iter() {
+        let WorldQueryFieldInfo {
+            field_type,
+            fetch_init_type: init_type,
+            is_ignored,
+            attrs,
+        } = read_world_query_field_info(field, world_lifetime, fetch_lifetime);
+
+        let field_ident = field.ident.as_ref().unwrap().clone();
+        if is_ignored {
+            ignored_field_attrs.push(attrs);
+            ignored_field_visibilities.push(field.vis.clone());
+            ignored_field_idents.push(field_ident.clone());
+            ignored_field_types.push(field.ty.clone());
+        } else {
+            field_attrs.push(attrs);
+            field_visibilities.push(field.vis.clone());
+            field_idents.push(field_ident.clone());
+            field_types.push(field_type);
+            fetch_init_types.push(init_type);
+        }
+    }
+
+    if fetch_struct_attributes.is_filter {
+        if has_world_lifetime {
+            panic!("Expected a struct without a lifetime");
+        }
```

Comment:
```suggestion
    if fetch_struct_attributes.is_filter && has_world_lifetime {
        panic!("Expected a struct without a lifetime");
```

Reply:
This change will make the macro panic when there's no `'world` lifetime for the filter struct. This is not something we want.